examples/android/binder/java/io/grpc/binder/cpp/exampleserver/native.cc

```diff
@@ -60,11 +58,10 @@ Java_io_grpc_binder_cpp_exampleserver_ExportedEndpointService_init_1grpc_1server
   grpc::ServerBuilder server_builder;
   server_builder.RegisterService(&service);
 
-  // TODO(mingcl): Uncomment this after server interfaces are merged
-  //
-  // grpc_endpoint_binder_pool_init();
-  // server_builder.AddListeningPort("binder://example.service",
-  // grpc::experimental::BinderServerCredentials());
+  grpc_endpoint_binder_pool_init();
```

Comment:
TaWei: What's the plan to avoid this call to init binder pool? I wonder why we can help user invoke this in ctor, as I comments before

Reply:
The problem is that I'm not sure how to initialize the mutex and the object it protects at the same time. `absl::Mutex` is not trivially destructible either (or is it?) so we can't have a static mutex for this purpose.

Reply:
> The problem is that I'm not sure how to initialize the mutex and the object it protects at the same time

Why is that an issue?

> absl::Mutex is not trivially destructible either (or is it?)

I think it is ok to intentionally leak it

Reply:
> Why is that an issue?

What I mean is that if both the mutex and the object are pointers that need to be heap-allocated for the first time we enter the constructor, then how are we gonna handle races that happens while we're initializing these two things? We can use an atomic integer to make sure they are only initialized once, but I'm not sure how to make others that come during the initialization wait until the initialization is over (of course we can have another atomic flag and all other threads just keep reading the flag until it is turned on, but it seems quite bad).

Reply:
Initialization of block-scoped static variable is thread safe (See https://www.modernescpp.com/index.php/thread-safe-initialization-of-data and https://stackoverflow.com/a/8102145)

Reply:
I think this is not the same issue I'm referring to... What I mean is that "if we don't want to use a static mutex", then we'd have to do something like
```
static absl::Mutex* mutex = nullptr;
static Object* object = nullptr;
if (this is the first time) {
  mutex = new absl::Mutex();  // what should we do when other thread runs this function while we're allocating mutex?
  object = new Object();
}
mutex->Lock();
...
```

But if having a static non-trivially-destructible mutex is acceptable to you, then yes, this can be easily handled.

Reply:
Discussed offline, TaWei will fix this later